### DISCUSSION TITLE
Where block is mismatching with existing where scope and eating one parameter

### DISCUSSION THREAD

**Discussion by madmaniak** [jeremyevans/sequel#2105] [General]
Hello,
I noticed weird behavior which I can't workaround. I have following piece of code:

```ruby
do
      f = from; t = to
      Slot.where{ (from >= f) & (to < t) }.count.zero?
end
```
I am in control of `Slot` and beneath we have `Sequel.model[:slots].where(removed: false)`. Context - `self` is a Sequel::Model Slot instance (not stored in DB yet). Above in result gives such query:

`
SELECT count(*) AS "count" FROM "slots" WHERE (("removed" IS FALSE) AND ("to" >= '2023-12-05 15:54:32.877326+0000')
`
As we see `from` parameter is missing - but not only that. The `to` parameter takes place of `from`...

Sequel 5.73

**Comment by jeremyevans** [comment]
I cannot reproduce:

```ruby
# run with bin/sequel -E sqlite:/
DB.create_table(:slots) do
  primary_key :id
  Time :from
  Time :to
  TrueClass :removed
end

class Slot < Sequel::Model(DB[:slots].where(removed: false))
end

t = Time.now
f = Time.now
Slot.where{ (from >= f) & (to < t) }.count.zero?
# SELECT count(*) AS 'count' FROM `slots`
# WHERE ((`removed` = 0) AND (`from` >= '2023-12-05 08:41:08.780932') AND (`to` < '2023-12-05 08:41:08.328756')) LIMIT 1
```

Can you please provide a self contained reproducible example?

**Comment by madmaniak** [comment]
Thanks a lot for quick reply. As I was struggling with it a while I solved it immediately after asking for help.
Problem was in Slot - which I am proxying through my model into Sequel::Model.
I did mismatched &blocks at some point.

**Comment by madmaniak** [comment]
```ruby
class Slot < Sequel::Model(DB[:slots].where(removed: false))
```

Very interesting line, writing down...
